src/probnum/diffeq/odefiltsmooth/_ivpfiltsmooth.py

```diff
@@ -231,12 +233,27 @@ def step(self, t, t_new, current_rv):
             errorest=local_errors,
             reference_state=reference_values,
         )
+
         if not self.steprule.is_accepted(internal_norm):
-            return np.nan * current_rv, local_errors, reference_values
+            t_new = state.t + dt
+            new_rv = randvars.Normal(
+                mean=state.rv.mean.copy(),
+                cov=state.rv.cov.copy(),
+                cov_cholesky=state.rv.cov_cholesky.copy(),
+            )
+            state = _odesolver_state.ODESolverState(
+                ivp=state.ivp,
+                rv=np.nan * new_rv,
+                t=t_new,
+                error_estimate=local_errors,
+                reference_state=reference_values,
+            )
+
+            return state
 
         # Now we can be certain that the step will be accepted. In this case
         # we update the diffusion model and continue the rest of the iteration.,
-        self.diffusion_model.update_in_place(local_diffusion, t=t)
+        self.diffusion_model.update_in_place(local_diffusion, t=state.t)
```

Comment:
Should this here be t, or t_new?

Reply:
I think it should be `t`. But I am not 100% sure (which is reinforcement that the diffusion model implementation is not at its final stage yet - it should be clear!).